src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -141,8 +146,19 @@ private function replaceConstructor(ClassMethod $method): void
             // Extract closures
             $bodyParts[] = '$this->extractCallbacks[] = \\Closure::bind(static function ($object, &$values) {';
             foreach ($properties as $property) {
-                $propertyName = $property->name;
-                $bodyParts[]  = "    \$values['" . $propertyName . "'] = \$object->" . $propertyName . ';';
+                $propertyName  = $property->name;
+                $requiresGuard = $property->hasType && ! ($property->hasDefault || $property->allowsNull);
+                $indent        = $requiresGuard ? '        ' : '    ';
+
+                if ($requiresGuard) {
+                    $bodyParts[] = '    if (isset($object->' . $propertyName . ')) {';
+                }
+
+                $bodyParts[]  = $indent . "\$values['" . $propertyName . "'] = \$object->" . $propertyName . ';';
+
+                if ($requiresGuard) {
+                    $bodyParts[] = '    }';
```

Comment:
not actually sure what the CS rules wanted me to fix here. Surely it doesn't want to invert the condition to put a continue here and then make the assignment outside of the if?

Reply:
Yeah, I think it wants a `continue;` or such - conditionals inside a `foreach` are (rightfully) frowned upon by the current CS rules

This is what's left:

```
Error: Equals sign not aligned correctly; expected 1 space but found 2 spaces
Error: Use early exit to reduce code nesting.
```

Reply:
That would be objectively worse code; I've extracted it into a method instead